modules/kargoBidAdapter.js

```diff
@@ -4,286 +4,484 @@ import { registerBidder } from '../src/adapters/bidderFactory.js';
+    if (crb && crb.v) {
+      let vParsed = JSON.parse(atob(crb.v));
+      if (vParsed) {
+        return vParsed;
       }
-      return {};
-    } catch (e) {
-      return {};
     }
-  },
+    return {};
+  } catch (e) {
+    return {};
+  }
+}
 
-  _getCrbFromLocalStorage() {
-    try {
-      return JSON.parse(atob(spec._getLocalStorageSafely('krg_crb')));
-    } catch (e) {
-      return {};
-    }
-  },
+function getCrbFromLocalStorage() {
+  try {
+    return JSON.parse(atob(getLocalStorageSafely(CERBERUS.KEY)));
+  } catch (e) {
+    return {};
+  }
+}
 
-  _getCrb() {
-    let localStorageCrb = spec._getCrbFromLocalStorage();
-    if (Object.keys(localStorageCrb).length) {
-      return localStorageCrb;
-    }
-    return spec._getCrbFromCookie();
-  },
-
-  _getLocalStorageSafely(key) {
-    try {
-      return storage.getDataFromLocalStorage(key);
-    } catch (e) {
-      return null;
-    }
-  },
-
-  _getUserIds(tdid, usp, gdpr) {
-    const crb = spec._getCrb();
-    const userIds = {
-      kargoID: crb.lexId,
-      clientID: crb.clientId,
-      crbIDs: crb.syncIds || {},
-      optOut: crb.optOut,
-      usp: usp
-    };
 
-    try {
-      if (gdpr) {
-        userIds['gdpr'] = {
-          consent: gdpr.consentString || '',
-          applies: !!gdpr.gdprApplies,
-        }
+
+function getLocalStorageSafely(key) {
+  try {
+    return STORAGE.getDataFromLocalStorage(key);
+  } catch (e) {
+    return null;
+  }
+}
+
+function getUserIds(tdid, usp, gdpr, eids) {
+  const crb = spec._getCrb();
+  const userIds = {
+    crbIDs: crb.syncIds || {}
+  };
+
+  if (tdid) {
+    userIds.tdID = tdid;
+  }
+
+  if(usp) {
+    userIds.usp = usp
+  }
+
+  try {
+    if (gdpr) {
+      userIds['gdpr'] = {
+        consent: gdpr.consentString || '',
+        applies: !!gdpr.gdprApplies,
       }
-    } catch (e) {
     }
-    if (tdid) {
-      userIds.tdID = tdid;
+  } catch (e) {
+  }
+  
+  if(crb.lexId != null) {
+    userIds.kargoID = crb.lexId
+  }
+
+  if(crb.clientId !=  null) {
+    userIds.clientID = crb.clientId
+  }
+
+  if(crb.optOut != null) {
+    userIds.optOut = crb.optOut
+  }
+
+  if(eids != null) {
+    userIds.eids = eids
+  }
+
+  return userIds;
+}
+
+function getClientId() {
+  const crb = spec._getCrb();
+  return crb.clientId;
+}
+
+function getAllMetadata(bidderRequest) {
+  return {
+    pageURL: bidderRequest?.refererInfo?.page,
+    rawCRB: STORAGE.getCookie(CERBERUS.KEY),
+    rawCRBLocalStorage: getLocalStorageSafely(CERBERUS.KEY)
+  };
+}
+
+function getRequestCount() {
+  if (lastPageUrl === window.location.pathname) {
+    return ++requestCounter;
+  }
+  lastPageUrl = window.location.pathname;
+  return requestCounter = 0;
+}
+
+function sendTimeoutData(auctionId, auctionTimeout) {
+  let params = {
+    aid: auctionId,
+    ato: auctionTimeout,
+  };
+
+  try {
+    let timeoutRequestUrl = buildUrl({
+      protocol: 'https',
+      hostname: 'krk.kargo.com',
```

Comment:
i might be incorrect but should this be using `krk2`?

Reply:
As far as I know the change in host in the Jira only relates to the bid requests call. Maybe @ssadman22 can clarify further?

Reply:
Hey @njflynn, `krk2` uses the same code as as `krk` but with added support for the HTTP2 protocol. I think we can use the new host `'krk2.kargo.com' for consistency and leverage its features

Reply:
Thanks @ssadman22 I have updated to krk2 now